### ISSUE TITLE
TS: Property 'createUserContext' is missing in type 'OptimizelyReactSDKClient' but required in type 'ReactSDKClient' (v2.4.2)

### ISSUE THREAD

**Issue by Bielousov** [optimizely/react-sdk#101]
Getting a TS error when adding `OptimizelyProvider` as per guide:

> TypeScript error in /src/index.tsx(32,17):
Property 'createUserContext' is missing in type 'OptimizelyReactSDKClient' but required in type 'ReactSDKClient'.  TS2741

```
    30 |             <ThemeProvider theme={theme}>
    31 |               <OptimizelyProvider
  > 32 |                 optimizely={optimizely}
       |                 ^
    33 |                 user={{
    34 |                   id: 'user123',
    35 |                 }}
```

React App code:

```
import { createInstance,  OptimizelyProvider } from '@optimizely/react-sdk';

const optimizely = createInstance({
  sdkKey: 'XXXXXXX',
});

ReactDOM.render(
  <React.StrictMode>
    <OptimizelyProvider
        optimizely={optimizely}
        user={{ id: 'user123' }}
      >
        <App />
      </OptimizelyProvider>
  </React.StrictMode>,
  document.getElementById('root'),
);
```

**Comment by BrucePauker** [comment]
Hello,

We have the same issue and it's blocking all the CI/CD because the build is not working anymore. Do you have a solution ?

Thanks !

**Comment by mikecousins** [comment]
We're seeing this as well.

**Comment by dtothefp** [comment]
Same issue

**Comment by remisture** [comment]
Same problem here

**Comment by zashraf1985** [comment]
This is happening because this SDK depends on [Javascript SDK](https://www.npmjs.com/package/@optimizely/optimizely-sdk) and the latest minor version of that does not appear to be compatible. While we fix it, an immediate workaround is to force the resolution of `Javascript SDK` to the compatible minor version i.e `4.4.3`. You can copy and paste the following in your package.json or add to the list of forced resolution lists if you already have any and it should work for now.
```
"resolutions": {
  "@optimizely/optimizely-sdk": "4.4.3"
},
```

**Comment by zashraf1985** [comment]
This is fixed in version [2.4.3](https://github.com/optimizely/react-sdk/releases/tag/2.4.3)
